src/common/w25x.c

```diff
@@ -1,130 +1,135 @@
-// static const uint8_t _CMD_ENABLE_WR_VSR = 0x50;
-static const uint8_t _CMD_DISABLE_WR = 0x04;
-static const uint8_t _CMD_RD_STATUS_REG = 0x05;
-static const uint8_t _CMD_WR_STATUS_REG = 0x01;
-static const uint8_t _CMD_RD_DATA = 0x03;
-// static const uint8_t _CMD_RD_FAST = 0x0b;
-// static const uint8_t _CMD_RD_FAST_D_O = 0x3b;
-// static const uint8_t _CMD_RD_FAST_D_IO = 0xbb;
-static const uint8_t _CMD_PAGE_PROGRAM = 0x02;
-static const uint8_t _CMD_SECTOR_ERASE = 0x20;
-static const uint8_t _CMD_BLOCK32_ERASE = 0x52;
-static const uint8_t _CMD_BLOCK64_ERASE = 0xd8;
-static const uint8_t _CMD_CHIP_ERASE = 0xc7;
-// static const uint8_t _CMD_CHIP_ERASE2 = 0x60;
-// static const uint8_t _CMD_PWR_DOWN = 0xb9;
-// static const uint8_t _CMD_PWR_DOWN_REL = 0xab;
-static const uint8_t _CMD_MFRID_DEVID = 0x90;
-// static const uint8_t _CMD_MFRID_DEVID_D = 0x92;
-// static const uint8_t _CMD_JEDEC_ID = 0x9f;
-static const uint8_t _CMD_RD_UID = 0x4b;
-
-#define _CS_LOW()  HAL_GPIO_WritePin(GPIOD, GPIO_PIN_7, 0)
-#define _CS_HIGH() HAL_GPIO_WritePin(GPIOD, GPIO_PIN_7, 1)
-
-static inline uint8_t spi_txrx(uint8_t tx);
-
-#define _SPI_TX(b) spi_txrx(b)
-#define _SPI_RX()  spi_txrx(0xff)
-
-extern SPI_HandleTypeDef hspi3;
-
-static inline uint8_t spi_txrx(uint8_t tx) {
-    uint8_t _in = 0;
-    HAL_SPI_TransmitReceive(&hspi3, &tx, &_in, 1, HAL_MAX_DELAY);
-    return _in;
-}
-
-int w25x_mfrid_devid(void);
-
-int8_t w25x_init(void) {
-    //PIN_OUT(W25X20CL_PIN_CS);
-    _CS_HIGH();
-    W25X_SPI_ENTER();
-    if (!w25x_mfrid_devid())
-        return 0;
-    return 1;
-}
-
-void w25x_enable_wr(void) {
-    _CS_LOW();
-    _SPI_TX(_CMD_ENABLE_WR); // send command 0x06
-    _CS_HIGH();
+#include "FreeRTOS.h"
+#include "w25x_communication.h"
+
+LOG_COMPONENT_DEF(W25X, LOG_SEVERITY_DEBUG);
+
+static const uint16_t PAGE_SIZE = 256;
+
+static const uint8_t MFRID = 0xEF;
+static const uint8_t DEVID = 0x13;
+static const uint8_t DEVID_NEW = 0x16;
+
+enum {
+    CMD_ENABLE_WR = 0x06,
+    CMD_ENABLE_WR_VSR = 0x50,
+    CMD_DISABLE_WR = 0x04,
+    CMD_RD_STATUS_REG = 0x05,
+    CMD_WR_STATUS_REG = 0x01,
+    CMD_RD_DATA = 0x03,
+    CMD_RD_FAST = 0x0b,
+    CMD_RD_FAST_D_O = 0x3b,
+    CMD_RD_FAST_D_IO = 0xbb,
+    CMD_PAGE_PROGRAM = 0x02,
+    CMD_SECTOR_ERASE = 0x20,
+    CMD_BLOCK32_ERASE = 0x52,
+    CMD_BLOCK64_ERASE = 0xd8,
+    CMD_CHIP_ERASE = 0xc7,
+    CMD_CHIP_ERASE2 = 0x60,
+    CMD_PWR_DOWN = 0xb9,
+    CMD_PWR_DOWN_REL = 0xab,
+    CMD_MFRID_DEVID = 0x90,
+    CMD_MFRID_DEVID_D = 0x92,
+    CMD_JEDEC_ID = 0x9f,
+    CMD_RD_UID = 0x4b,
+};
+
+typedef enum {
+    W25X_STATUS_BUSY = 0x01,
+    W25X_STATUS_WEL = 0x02,
+    W25X_STATUS_BP0 = 0x04,
+    W25X_STATUS_BP1 = 0x08,
+    W25X_STATUS_TB = 0x20,
+    W25X_STATUS_SRP = 0x80,
+} w25x_status_t;
+
+static inline void cs_low() {
+    HAL_GPIO_WritePin(GPIOD, GPIO_PIN_7, 0);
+}
+
+static inline void cs_high() {
+    HAL_GPIO_WritePin(GPIOD, GPIO_PIN_7, 1);
+}
```

Comment:
Pin is configured in hwio_pindef.h with symbolic name, but here hardcoded pin number is used.

Reply:
Fixed